Add needles for selecting the TPM2+PIN method

mtravitzky opened the pull request:
Add needles for selecting the TPM2+PIN method in jeos-firstboot. Additionally, needle file names have been fixed (missing dash before date).

Vogtinator commented on encrypted-disk-password-prompt-tpm-20250507.json:
```diff
@@ -0,0 +1,15 @@
+  ],
+  "properties": [],
+  "tags": [
+    "encrypted-disk-password-prompt"
```
Should probably be a different tag to show PIN vs. password entry

mtravitzky replied:
Fixed.

Vogtinator reviewed (changes requested):
Needs a separate tag

mtravitzky commented:
> We normally avoid having merge commits on feature branches. Can you just rebase against master? It would perhaps also make sense to squash the commits.
> 
> And do you need help merging this (as this PR has been sitting here now for 3 weeks)?

Rebased and squashed. Should be okay to merge now.
